model/registry/cloudfoundry.yaml

```diff
@@ -0,0 +1,158 @@
+groups:
+  - id: registry.cloudfoundry
+    prefix: cloudfoundry
+    type: attribute_group
+    brief: >
+      CloudFoundry resource attributes.
+    attributes:
+      - id: system.id
+        type: string
+        stability: experimental
+        brief: >
+          A guid or another name describing the event source.
+        note: |
+          CloudFoundry defines the `source_id` in the [Loggegator v2 envelope](https://github.com/cloudfoundry/loggregator-api#v2-envelope).
+          It is used for logs and metrics emitted by CloudFoundry. It is
+          supposed to contain the component name, e.g. "gorouter", for
+          CloudFoundry components.
+
+          When system components are instrumented, values from the
+          [Bosh spec](https://bosh.io/docs/jobs/#properties-spec)
+          should be used. The `system.id` should be set to
+          `spec.deployment/spec.name`.
+        examples:
+          - 'cf/gorouter'
+      - id: system.instance.id
+        type: string
+        stability: experimental
+        brief: >
+          A guid describing the concrete instance of the event source.
+        note: |
+          CloudFoundry defines the `instance_id` in the [Loggegator v2 envelope](https://github.com/cloudfoundry/loggregator-api#v2-envelope).
+          It is used for logs and metrics emitted by CloudFoundry. It is
+          supposed to contain the vm id for CloudFoundry components.
+
+          When system components are instrumented, values from the
+          [Bosh spec](https://bosh.io/docs/jobs/#properties-spec)
+          should be used. The `system.instance.id` should be set to `spec.id`.
+        examples:
+          - '218fc5a9-a5f1-4b54-aa05-46717d0ab26d'
+      - id: app.name
+        type: string
+        stability: experimental
+        brief: >
+          The name of the application.
+        note: |
+          Application instrumentation should use the value from environment
+          variable `VCAP_APPLICATION.application_name`. This is the same value
+          as reported by `cf apps`.
+        examples: ['my-app-name']
+      - id: app.id
+        type: string
+        stability: experimental
+        brief: >
+          The guid of the application.
+        note: |
+          Application instrumentation should use the value from environment
+          variable `VCAP_APPLICATION.application_id`. This is the same value as
+          reported by `cf app <app-name> --guid`.
+        examples: ['218fc5a9-a5f1-4b54-aa05-46717d0ab26d']
+      - id: app.instance.id
```

Comment:
maybe this one is similar to `service.instance.id` ?

Reply:
CloudFoundry does not use unique guid or each app instance. Instead, a single guid is created for the entire instance group known as `app.id`. If you run multiple instances of the app, they will have different instance indices as explained below. This index is captured in `app.instance.id`.

This entire approach differs from OpenTelemetry's [service attributes](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/service.md). Therefore, different fields were used.

Maybe it would be clearer if `app.instance.id` was renamed to `app.instance.index`? What do you think?